server/index.js

```diff
@@ -0,0 +1,231 @@
+  email: joi.string().email().trim().lowercase().required(),
+  password: joi.string().required(),
+});
+const dbPool = mysql.createPool(mysqlConfig).promise();
+
+server.get("/groups", authenticate, async (req, res) => {
+  console.log(req.user);
+
+  try {
+    const [data] = await dbPool.execute(
+      `
+      SELECT * FROM  \`groups\`
+    `,
+    );
+    return res.status(201).send({ data });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).end();
+  }
+});
+
+server.post("/groups", authenticate, async (req, res) => {
+  let payload = req.body;
+
+  try {
+    await dbPool.execute(
+      `
+  INSERT INTO \`groups\` (id, name)
+  VALUES (?,?)`,
+      [payload.id, payload.name],
+    );
+    return res.status(201).end();
+  } catch (error) {
+    console.error(error);
+    return res.status(500).end();
+  }
+});
+
+server.post("/login", async (req, res) => {
+  let payload = req.body;
+
+  // Validation
+  const loginSchema = joi.object({
+    email: joi.string().email().trim().lowercase().required(),
+    password: joi.string().required(),
+  });
+
+  try {
+    payload = await loginSchema.validateAsync(payload);
+  } catch (error) {
+    console.error(error);
+    return res.status(400).send({ error: "All fields are required" });
+  }
+  // Checking if email exists
+
+  try {
+    const [data] = await dbPool.execute(
+      `
+    SELECT * FROM users
+    WHERE email = ?
+    `,
+      [payload.email],
+    );
+
+    if (!data.length) {
+      return res.status(400).send({ error: "Email or Password did not match" });
+    }
+
+    const isPasswordMatching = await bcrypt.compare(
+      payload.password,
+      data[0].password,
+    );
+
+    if (isPasswordMatching) {
+      const token = jwt.sign(
+        {
+          email: data[0].email,
+          id: data[0].id,
+        },
+        process.env.JWT_SECRET,
+      );
+      return res.status(200).send({ token });
+    }
+    return res.status(400).send({ error: "Email or Password did not match" });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).end();
+  }
+});
+
+server.post("/register", async (req, res) => {
+  let payload = req.body;
+
+  try {
+    payload = await userSchema.validateAsync(payload);
+  } catch (error) {
+    console.error(error);
+    return res.status(400).send({ error: "All fields are required" });
+  }
+
+  try {
+    const encryptedPassword = await bcrypt.hash(payload.password, 10);
+    await dbPool.execute(
+      `
+        INSERT INTO users (full_name, email, password)
+        VALUES (?, ?, ?)
+        `,
+      [payload.full_name, payload.email, encryptedPassword],
+    );
+    return res.status(201).end();
+  } catch (error) {
+    console.error(error);
+    return res.status(500).end();
+  }
+});
+
+server.post("/accounts", authenticate, async (req, res) => {
+  let payload = req.body;
+
+  const token = req.headers.authorization?.split(" ")[1];
+  const decoded = jwt_decode(token);
```

Comment:
You should take user info from `req.user` as you are assigning users details in the middleware. By doing so you won't need to get token and decode it in here